features/messages/impl/src/test/kotlin/io/element/android/features/messages/textcomposer/MessageComposerPresenterTest.kt

```diff
@@ -285,16 +297,83 @@ class MessageComposerPresenterTest {
     }
 
     @Test
-    fun `present - Pick media from gallery`() = runTest {
-        val presenter = createPresenter(this)
+    fun `present - Pick image from gallery`() = runTest {
+        val room = FakeMatrixRoom()
+        val presenter = createPresenter(this, room = room)
         pickerProvider.givenMimeType(MimeTypes.Images)
+        mediaPreProcessor.givenResult(Result.success(
+            MediaUploadInfo.Image(
+                file = File("/some/path"),
+                info = ImageInfo(
+                    width = null,
+                    height = null,
+                    mimetype = null,
+                    size = null,
+                    thumbnailInfo = null,
+                    thumbnailUrl = null,
+                    blurhash = null,
+                ),
+                thumbnailInfo = ThumbnailProcessingInfo(
+                    file = File("/some/path"),
+                    info = ThumbnailInfo(
+                        width = null,
+                        height = null,
+                        mimetype = null,
+                        size = null,
+                    ),
+                    blurhash = "",
+                )
+            )
+        ))
         moleculeFlow(RecompositionClock.Immediate) {
             presenter.present()
         }.test {
             val initialState = awaitItem()
             initialState.eventSink(MessageComposerEvents.PickAttachmentSource.FromGallery)
+            // Wait for the launched upload coroutine to run
+            delay(1)
```

Comment:
What about using `TestScope.runCurrent()` instead of delaying with a "random" value? It's seems more accurate to me but not a big deal.

Reply:
That's way better! I knew there was something like that but I couldn't remember its name. Thanks!